### PR TITLE
Describe doc

### LINKED ISSUE #3265: Improving docstring wrapping functions
We often reuse NumPy functions' docstrings. This is convenient as we don't take up maintaining docstrings (or more likely allowing them to go stale) when we don't have anything new to add. However there are few downsides to this approach (and possibly others I'm unaware of).

1. NumPy versions show up (very confusing to users). ( https://github.com/ioam/holoviews/pull/2424#issuecomment-371896217 )
2. Arguments show up that are not fully or at all supported (also very confusing to users). ( https://github.com/dask/dask/issues/2946 )
3. Examples use NumPy instead of Dask.

Mainly these sorts of things confuse new users and leave them wondering about what Dask actually supports. This is not a great situation for us to be in. There are a few ways we can improve this.

1. Manually copy and tweak docstrings when they deviate.
2. Scrub things in our wrapper function (e.g. `versionadded`, missing args, examples).
3. Add a note to the docs pointing out how the docstrings are made.
4. Add any missing behavior (though still need to handle version added).

This could be an issue with Pandas too. I don't work with that part of the code enough to know.

Thoughts on this?

### LINKED ISSUE #4401: Dask dataframe 'describe()' option fails for 'object' dtype
If I look at the documentation: http://docs.dask.org/en/latest/dataframe-api.html, it seems to be that 'object' dtypes should be accepted with the 'describe()' option by means of this sentence: "For object data (e.g. strings or timestamps), the result’s index will include count, unique, top, and freq", but as I can see in one of my examples, it does not work yet, giving the following error message: "ValueError: DataFrame contains only non-numeric data."
Any news about it so far? Thanks in advance.

### PR DESCRIPTION
**Pull request by martindurant**
- [ ] Tests added / passed
- [x] Passes `flake8 dask`

Written to add extra information to the derived docstring for dataframe.describe ( fixes #4401 )

partially fixes #3265 (simplest solution)

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -1537,7 +1537,7 @@ def quantile(self, q=0.5, axis=0, method='default'):
 
     @derived_from(pd.DataFrame)
     def describe(self, split_every=False, percentiles=None, percentiles_method='default'):
-        # currently, only numeric describe is supported
+        """Currently, only numeric describe is supported """
```
Comment (mrocklin) [inline dask/dataframe/core.py]: We might use this function as a test for the derived_from functionality change in this PR. 

As an example, I think that there is already a test in `test_utils.py` that import dask.dataframe and checks the state of one of the docstrings.

Comment (martindurant) [reply dask/dataframe/core.py]: good idea

File: dask/utils.py [hunk 2]
```diff
@@ -486,12 +486,16 @@ def ignore_warning(doc, cls, name):
...
+        bits = [
             head,
             indent, l1,
             indent, l2, '\n\n',
             tail
-        ])
+        ]
+        if extra:
+            bits.insert(1, extra.rstrip('\n') + '\n\n')
+            bits.insert(1, indent)
```
Comment (mrocklin) [inline dask/utils.py]: Would it be possible to move extra below the "this is copied" disclaimer?

```
Signature: dd.DataFrame.describe(self, split_every=False, percentiles=None, percentiles_method='default')
Docstring:
Generate descriptive statistics that summarize the central tendency,
dispersion and shape of a dataset's distribution, excluding
``NaN`` values.

Currently, only numeric describe is supported

This docstring was copied from pandas.core.frame.DataFrame.describe.

Some inconsistencies with the Dask version may exist.

Analyzes both numeric and object series, as well
as ``DataFrame`` column sets of mixed data types. The output
will vary depending on what is provided. Refer to the notes
below for more detail.
```

Comment (martindurant) [reply dask/utils.py]: done

**Comment by martindurant** [comment]
The errors are py2-only, and I really don't understand them or know if they are caused by this code (seem to be import errors of "methods"). Help!

**Comment by mrocklin** [comment]
> The errors are py2-only, and I really don't understand them or know if they are caused by this code (seem to be import errors of "methods"). Help!

The docstring munging code runs at import time (we have `@derived_from` decorators in the main file).  In your situation I would create a Python 2 environment, try importing the module, and see what happens.  Maybe I would also use the Python debugger (if possible at this stage) or maybe put in a pdb statement?

I'm not sure if that helps or not.  Hopefully those steps move towards a good solution.

**Comment by martindurant** [comment]
There we go :)  It was to do with docstring stripping during bytecode optimisation.

**Comment by martindurant** [comment]
I think this should be merged now, can press the button if no one objects by tomorrow morning

**Pull request merged**